types/jest/index.d.ts

```diff
@@ -126,7 +126,7 @@ declare namespace jest {
     /**
      * Mocks a module with an auto-mocked version when it is being required.
      */
-    function mock(moduleName: string, factory?: any, options?: MockOptions): typeof jest;
+    function mock<T extends {}>(moduleName: string, factory?: () => T, options?: MockOptions): typeof jest;
```

Comment:
Would

```ts
function mock<T extends string>(moduleName: T, factory?: () => typeof import(T), options?: MockOptions): this
```

work?

Reply:
`factory?: () => typeof import(T)` doesn't look to make sense. Have you seen it being used that way before?

Reply:
It won't work, the syntax is incorrect. `import` expect string literal as argument